src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,200 @@
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            itemsDeleted = statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't delete shopping cart with id = "
+                    + id, ex);
+        }
+
+        removeAllProductsFromShoppingCart(id);
+        return itemsDeleted == 1;
+    }
+
+    private ShoppingCart getCartFromResultSet(ResultSet resultSet) throws SQLException {
+        Long cartId = resultSet.getLong("cart_id");
+        Long userId = resultSet.getLong("user_id");
+        return new ShoppingCart(cartId, userId);
+    }
+
+    private List<Product> getListOfProducts(Long cartId) {
+        String query = "SELECT cp.product_id AS id, p.product_name AS name, p.price "
+                + "FROM carts_products cp "
+                + "INNER JOIN products p "
+                + "ON cp.product_id = p.product_id "
+                + "WHERE cp.cart_id = ?;";
+        List<Product> products = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, cartId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Long productId = resultSet.getLong("id");
+                String productName = resultSet.getString("name");
+                Long price = resultSet.getLong("price");
+                Product product = new Product(productId, productName, price);
+                products.add(product);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get products from shopping cart with id = "
+                    + cartId, ex);
+        }
+        return products;
+    }
+
+    private void removeAllProductsFromShoppingCart(Long cartId) {
+        String query = "DELETE FROM carts_products WHERE cart_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, cartId);
+            statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't remove products from shopping cart with id = "
+                    + cartId, ex);
+        }
+    }
+
+    private void insertProductsToShoppingCart(ShoppingCart cart) {
+        String query = "INSERT INTO carts_products (cart_id, product_id) VALUES (?, ?);";
+        for (Product product: cart.getProducts()) {
+            try (Connection connection = ConnectionUtil.getConnection()) {
+                PreparedStatement statement = connection.prepareStatement(query);
+                statement.setLong(1, cart.getId());
+                statement.setLong(2, product.getId());
+                statement.executeUpdate();
```

Comment:
You can use loop only for this 2 lines, so we will not open connections each time

Reply:
Thanks! I thought we should rebiuld statement for every loop iteration but because of your comment understood that it is unnecessary)